### PR TITLE
De-XHRify

### PR DESCRIPTION
**Pull request by wbamberg**
Part of https://github.com/openwebdocs/project/issues/156. I had to keep the event listeners as non-arrow functions, I think because they depend on the `this` value.

### CODE REVIEW

**Review by estelle — APPROVED** [review]
Looks good.

File: imscjs-demo/js/index.js
Comment (teoli2003) [inline imscjs-demo/js/index.js:21]: I don't think this is still needed: VTTCue is unlikely to be supported by Edge > 79. Anyway, `TextTrackCue` has been supported by every engine for ages; we can simplify this. (as well as the call to the `Cue()` that would be a `new TextTrackCue()`on line 25 and 30.
Capture d’écran 2023-12-20 à 06 42 49

File: imscjs-demo/js/index.js
Comment (teoli2003) [inline imscjs-demo/js/index.js:19]: Given that we modernize this example at the same time and as`timeEvents` is an `Array`. Can you double-check if we could use `for...of` here (and get rid of `i`? (Not 100% sure it is possible)

File: imscjs-demo/js/index.js
Comment (teoli2003) [inline imscjs-demo/js/index.js:29]:
```suggestion
        // "End" time of the last "imsc event" is the end of the video
```

File: imscjs-demo/js/index.js
Comment (teoli2003) [inline imscjs-demo/js/index.js:18]: What does isd mean?

```suggestion
    // Create one cue per timed event and render isd
```

File: imscjs-demo/js/index.js
Comment (teoli2003) [inline imscjs-demo/js/index.js:4]:
```suggestion
  // Track object generated by the first <track> child element
```

File: imscjs-demo/js/index.js
Comment (teoli2003) [inline imscjs-demo/js/index.js:6]:
```suggestion
  // The source is only available from the <track> DOM element, not the object
```

File: imscjs-demo/js/index.js
Comment (teoli2003) [inline imscjs-demo/js/index.js:9]:
```suggestion
  // The track is disabled by default, 'hidden' does not display VTT Text
  // but fires the events
```

File: imscjs-demo/js/index.js
Comment (teoli2003) [inline imscjs-demo/js/index.js:24]:
```suggestion
        // We have to provide an empty string as the VTTText
```

File: imscjs-demo/js/index.js
Comment (teoli2003) [inline imscjs-demo/js/index.js:52]: I don't know the advantages of prefixing this with `window`, I would use nothing or `self`, but I don't think this is a rule (maybe we should start a discussion at some point).
```suggestion
addEventListener("load", initTTML);
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Mostly nits, but we don't plan to reedit this file in the near future…

File: imscjs-demo/js/index.js
Comment (wbamberg) [reply imscjs-demo/js/index.js:19]: We use `i` at https://github.com/mdn/imsc-examples/blob/943d004dfac68e5df3aecd25da461be357415171/imscjs-demo/js/index.js#L29-L31 which is why I didn't do this.

File: imscjs-demo/js/index.js
Comment (wbamberg) [reply imscjs-demo/js/index.js:18]: no idea

Comment (teoli2003) [reply imscjs-demo/js/index.js]: Let's just fix the capitalization then.

Comment (wbamberg) [reply imscjs-demo/js/index.js]: "render it", maybe??

Comment (teoli2003) [reply imscjs-demo/js/index.js]: Yes, you are right!

File: imscjs-demo/js/index.js
Comment (wbamberg) [reply imscjs-demo/js/index.js]: As far as I can tell the correct thing here is to use `VTTCue`: it's supported everywhere. `TextTrackCue` is an abstract class, we couldn't create it.

File: imscjs-demo/js/index.js
Comment (teoli2003) [inline imscjs-demo/js/index.js:23]: I think the standard is `TextTrackCue`, doesn't it?

File: imscjs-demo/js/index.js
Comment (teoli2003) [inline imscjs-demo/js/index.js:18]:
```suggestion
    // Create one cue per timed event and render it
```

File: imscjs-demo/js/index.js
Comment (wbamberg) [reply imscjs-demo/js/index.js:23]: I know basically nothing about this API, but it looks to me like you can't construct a `TextTrackCue`, it doesn't have a constructor. From https://developer.mozilla.org/en-US/docs/web/api/texttrackcue:

> TextTrackCue is an abstract class which is used as the basis for the various derived cue types, such as [VTTCue](https://developer.mozilla.org/en-US/docs/Web/API/VTTCue); you will instead work with those derived types.

Comment (wbamberg) [reply imscjs-demo/js/index.js:23]: I have tested this demo in Firefox and Chrome with VTTCue and it works, and with TextTrackCue, and it doesn't.

Comment (teoli2003) [reply imscjs-demo/js/index.js:23]: 👍

**Review by teoli2003 — APPROVED** [review]
👍 Nice work. Let's wait for somebody with merge access, this is ready to be merged.

**Review by Userlugard — APPROVED** [review]

File: imscjs-demo/js/index.js
Comment (palemieux) [inline imscjs-demo/js/index.js:18]:
```suggestion
    // Since `TextTrackCue`/`VTTCue` does not support TTML natively, we create one (empty)
    // cue per event on the timeline of the TTML document. When the cue is triggered 
    // (`enter` event), we render the TTML document corresponding to the cue time, 
    // which is called an Intermediary Synchronic Document (ISD), into `renderDiv`.
```

**Comment by palemieux** [comment]
Just added a small suggestion clarifying terms

File: imscjs-demo/js/index.js
Comment (wbamberg) [reply imscjs-demo/js/index.js]: Great comment, thank you!

**Review by dipikabh — APPROVED** [review]
Approving this based on other reviews to check if I have merge rights. As @teoli2003 has indicated, this PR is ready to be merged.

**Pull request merged**
